ledger/src/blockstore_processor.rs

```diff
@@ -376,6 +402,7 @@ fn execute_batches(
                         &sanitized_txs,
                         slice_start,
                         index,
+                        &transaction_indexes,
```

Comment:
The rebatching respects the transaction index order right because there's no shuffle right (i.e. we only chop off the tail)? Might be good to augment the rebatching unit test to guarantee this